common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -214,35 +214,40 @@ protected boolean processRedirectUrl(@NonNull final WebView view, @NonNull final
 
     private boolean processWebsiteRequest(@NonNull final WebView view, @NonNull final String url) {
         final String methodName = "#processWebsiteRequest";
-        final PackageHelper packageHelper = new PackageHelper(getActivity().getPackageManager());
-        final Context applicationContext = getActivity().getApplicationContext();
-        if (url.startsWith(AuthenticationConstants.Broker.BROWSER_DEVICE_CA_URL)
-                && packageHelper.isPackageInstalledAndEnabled(applicationContext, AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME)
-                && packageHelper.isPackageInstalledAndEnabled(applicationContext, AuthenticationConstants.Broker.IPPHONE_APP_PACKAGE_NAME)
-                && AuthenticationConstants.Broker.IPPHONE_APP_SIGNATURE.equals(
-                packageHelper.getCurrentSignatureForPackage(AuthenticationConstants.Broker.IPPHONE_APP_PACKAGE_NAME))) {
```

Comment:
Why is this check removed ? I see a TODO, but did we test this ?

Reply:
Once we get a clear signal, we can remove this app-specific workaround.

I've tested this by adding ismdmurl=1 at the end of the url and set COMPANY_PORTAL_APP_PACKAGE_NAME to CP package in debug mode, CP is launched.

Reply:
(if we use brokertestapp, then an exception will be thrown as it wouldn't be able to find SplashActivity. In that case we'll proceed to the next step - launching browser and returning AuthenticationConstants.UIResponse.BROWSER_CODE_MDM.

Reply:
okay makes sense, probably this check is the reason, we always end up in Playstore even if CP is installed

Reply:
I guess this check was probably added because playstore is not available on ipphone devices. 

Otherwise the user can redirect to the app through playstore. (one more click -but it's not blocking).

Reply:
Yup, never made sense to me,  as why not directly go to CP in regular case also, i.e. why a specific app check. anyways this should fix